### PR TITLE
target: get model form platform

### PR DESCRIPTION
**Pull request by setrofim**
Move the resolution of the model name from targets into Platform's
_set_model_from_target() (which was already attempting to do that via
dmidecode method).

### CODE REVIEW

File: devlib/platform/__init__.py [hunk 1]
```diff
@@ -78,7 +78,16 @@ def _set_core_clusters_from_core_names(self):
 
     def _set_model_from_target(self, target):
         if target.os == 'android':
-            self.model = target.getprop('ro.product.model')
+            try:
+                self.model = self.getprop(prop='ro.product.device')
```
Comment (marcbonnici) [inline devlib/platform/__init__.py]: Should this not be `target.getprop()`?

File: devlib/platform/__init__.py [hunk 2]
```diff
@@ -78,7 +78,16 @@ def _set_core_clusters_from_core_names(self):
...
+            except KeyError:
+                self.model = target.getprop('ro.product.model')
+        elif self.file_exists("/proc/device-tree/model"):
+            # There is currently no better way to do this cross platform.
+            # ARM does not have dmidecode
+            raw_model = self.execute("cat /proc/device-tree/model")
```
Comment (marcbonnici) [inline devlib/platform/__init__.py]: `target.execute()`

File: devlib/platform/__init__.py [hunk 3]
```diff
@@ -78,7 +78,16 @@ def _set_core_clusters_from_core_names(self):
 
     def _set_model_from_target(self, target):
         if target.os == 'android':
-            self.model = target.getprop('ro.product.model')
+            try:
+                self.model = self.getprop(prop='ro.product.device')
+            except KeyError:
+                self.model = target.getprop('ro.product.model')
+        elif self.file_exists("/proc/device-tree/model"):
```
Comment (marcbonnici) [inline devlib/platform/__init__.py]: `target.file_exists()`

File: devlib/platform/__init__.py (hunk 1 above)
Comment (setrofim) [reply devlib/platform/__init__.py]: Yup it should. Thanks for catching my stupidity. Should be fixed now.

**Pull request merged**